### PR TITLE
Update revpi to kernel 5.10

### PR DESCRIPTION
**Pull request by chrblabla**
In the process of adding support for the Kunbus Revolution Pi Connect S (see #913), we understood, that we need to update the kernel to 5.10 first. This is being done with this PR.

Compilation was successful, but testing on a real device is still to be done. We can do this beginning of next week and will share the results here.

I'll also update the commit messages to match the required standards as mentioned in https://github.com/balena-os/balena-raspberrypi#pull-requests.

### CODE REVIEW

**Comment by nbuchwitz** [comment]
I don't have an, experience with the balena build process, but kernel refs and updated patches LGTM. Make sure to use the latest version of the dtoverlays as well (don't know if they are build with the kernel or separately in balena)

**Comment by chrblabla** [comment]
@nbuchwitz I just flashed a RevPi Connect with this build and it works beautifully.

**Comment by chrblabla** [comment]
@nbuchwitz one thing we observed is, that it now shows three ethernet interfaces (`eth0`, `eth1`, `eth2`). @floion can you point me towards updated device trees?

**Comment by chrblabla** [comment]
@nbuchwitz now having the kernel 5.10.120 running for a day on a RevPi Connect, we continue to see ethernet failures as described here: https://revolutionpi.de/forum/viewtopic.php?p=12904&hilit=ethernet#p12904

Any idea on how to advance here?

We would also try to get our hands onto another RevPi to rule out any hardware-related issues.

**Comment by nbuchwitz** [comment]
@chrblabla are you using the right devicetree (build from the kernel tag) overlay? If the ethernet adapter stays off this is usually related to missing USB driver. The host USB controller in Compute Module 4S in disabled per default, so you need to activate it in the devicetree. This is fixed in upstream with https://github.com/raspberrypi/linux/pull/5140 and also in our kernel repo (but not yet included in the last tagged version, thus needs to be done in the overlay).

**Comment by chrblabla** [comment]
@nbuchwitz thank you for pointing us in this direction, we'll build another image and let you know!

**Comment by floion** [comment]
Hi @chrblabla can you squash the first 3 commits into just 1 commit please?

**Comment by floion** [comment]
@resin-jenkins test this please

**Comment by floion** [comment]
@chrblabla @nbuchwitz the overlays are built from the kernel tree in balena also so having the kernel updated means the updated overlays are being compiled too.
@chrblabla let me know if you see something odd with the generated overlays

**Comment by chrblabla** [comment]
@floion I squashed the commits, should be good to go now.

**Comment by wickywaka** [comment]
Just commenting here so that we don't forget it: There is a new release of the kunbus linux kernel as well. It be seen here:
https://github.com/RevolutionPi/linux/tags

**Comment by floion** [comment]
@wickywaka can you update it to the latest kunbus kernel? And also rebase the PR in order to remove the merge commit?

@chrblabla @nbuchwitz  why do we need to do the interface renaming? Does any other component depend on a specific name for it?

**Comment by nbuchwitz** [comment]
@floion the interface is renamed in order to make it clear that this is not a normal ethernet interface, but art of the PiBridge expansion bus. I'm also not 100% sure if the kernel driver piControl relies on the name schema

**Comment by floion** [comment]
Thank for the additional info @nbuchwitz 
Do you have the source code of your OS images available to share? Or just the rootfs?

**Comment by nbuchwitz** [comment]
@floion everything is available. The kernel driver is located in the piControl repo, th udev stuff is handled in the package Revpi-tools (repo name is the same). If you need more details please let me know

**Comment by wickywaka** [comment]
> @wickywaka can you update it to the latest kunbus kernel? And also rebase the PR in order to remove the merge commit?

Did the squashing. Will update the kernel tag to latest soon.

**Comment by wickywaka** [comment]
@nbuchwitz If someone from your team want to support us in this merge request, you can tag them here and I will give them write access to the repository. That way they will be able to push into this branch.

File: layers/meta-balena-raspberrypi/recipes-kernel/picontrol/picontrol.bb [hunk 1]
```diff
@@ -10,8 +10,7 @@ SRC_URI = " \
 	file://0002-Search-config-file-in-mnt-boot.patch \
 "
 
-# Corresponds to tag raspberrypi-kernel_9.20200616-4.19.95+revpi1
-SRCREV ="72a58797b2eed6edab99795476108cfeec95c8d9"
+SRCREV ="raspberrypi-kernel_1%9.20220728-5.10.120+revpi1"
```
Comment (chrblabla) [inline layers/meta-balena-raspberrypi/recipes-kernel/picontrol/picontrol.bb:13]: @wickywaka can you please update this to `raspberrypi-kernel_1%9.20221118-5.10.152+revpi1`, too?

Comment (wickywaka) [reply layers/meta-balena-raspberrypi/recipes-kernel/picontrol/picontrol.bb]: > @wickywaka can you please update this to `raspberrypi-kernel_1%9.20221118-5.10.152+revpi1`, too?

Done.

**Comment by wickywaka** [comment]
@floion @nbuchwitz 
The Kernel and PiControl are both now the latest version and it seems to be compiling fine.

Can you guys have a look and see if something is missing?

Also I see this error (on multiple local branches) when I compile more that 3 times, any idea why is that happening?
```
ERROR: balena-image-1.0-r0 do_image_complete: ExecutionError('/home/ubuntu/balena-raspberrypi/build/tmp/work/revpi_connect_s-poky-linux/balena-image/1.0-r0/temp/run.deploy_image_license_manifest.3114489', 1, None, None)
ERROR: Logfile of failure stored in: /home/ubuntu/balena-raspberrypi/build/tmp/work/revpi_connect_s-poky-linux/balena-image/1.0-r0/temp/log.do_image_complete.3114489
Log data follows:
| DEBUG: Executing python function sstate_task_prefunc
| DEBUG: Python function sstate_task_prefunc finished
| DEBUG: Executing python function do_image_complete
| NOTE: Executing deploy_image_license_manifest ...
| DEBUG: Executing shell function deploy_image_license_manifest
| cp: cannot stat '/home/ubuntu/balena-raspberrypi/build/tmp/deploy/licenses/balena-image-revpi-connect-s-20230120061525/image_license.manifest': No such file or directory
| WARNING: exit code 1 from a shell command.
| DEBUG: Python function do_image_complete finished
ERROR: Task (/home/ubuntu/balena-raspberrypi/build/../layers/meta-balena/meta-balena-common/recipes-core/images/balena-image.bb:do_image_complete) failed with exit code '1'
NOTE: Tasks Summary: Attempted 4447 tasks of which 4413 didn't need to be rerun and 1 failed.

```

File: layers/meta-balena-raspberrypi/recipes-kernel/picontrol/picontrol/0001-Use-modules_install-as-wanted-by-yocto.patch [hunk 2]
```diff
@@ -15,14 +15,14 @@ index 56eab93..6d78b95 100644
 +++ b/Makefile
 @@ -44,6 +44,9 @@ EXTRA_CFLAGS += -D__KUNBUSPI_KERNEL__
  all: compiletime.h
- 	$(MAKE) ARCH=arm CROSS_COMPILE=arm-linux-gnueabihf- -C $(KDIR) M=$(PWD)  modules
- 
+ 	$(MAKE) ARCH=arm CROSS_COMPILE="$(CROSS_COMPILE)" -C $(KDIR) M=$(PWD)  modules
+
 +modules_install:
-+	$(MAKE) ARCH=$(ARCH) CROSS_COMPILE=$(CROSS_COMPILE) -C $(KDIR) M=$(shell pwd)  modules_install
```
Comment (floion) [inline layers/meta-balena-raspberrypi/recipes-kernel/picontrol/picontrol/0001-Use-modules_install-as-wanted-by-yocto.patch:21]: Is this patch really necessary? Do we get an error compiling the kernel if we don't change this patch? In my other PR with the kernel update I think these changes were not needed

File: layers/meta-balena-raspberrypi/recipes-kernel/linux-kunbus/linux-kunbus.bb [hunk 3]
```diff
@@ -1,8 +1,8 @@
 FILESEXTRAPATHS:prepend := "${THISDIR}/${PN}:"
 
-LINUX_VERSION = "5.10.120"
+LINUX_VERSION = "5.10.152"
 
-SRCREV = "raspberrypi-kernel_1%9.20220728-5.10.120+revpi1"
```
Comment (floion) [inline layers/meta-balena-raspberrypi/recipes-kernel/linux-kunbus/linux-kunbus.bb]: Can you add these changes in the first commit please? Here I mean https://github.com/balena-os/balena-raspberrypi/pull/917/commits/ba0e2bc3a7e1f6f7d99741cdc39ac24da0709782
So we don't have multiple commits that change twice the kernel revision and keeps the git history cleaner

**Comment by floion** [comment]
> @floion @nbuchwitz The Kernel and PiControl are both now the latest version and it seems to be compiling fine.
> 
> Can you guys have a look and see if something is missing?
> 
> Also I see this error (on multiple local branches) when I compile more that 3 times, any idea why is that happening?
> 
> ```
> ERROR: balena-image-1.0-r0 do_image_complete: ExecutionError('/home/ubuntu/balena-raspberrypi/build/tmp/work/revpi_connect_s-poky-linux/balena-image/1.0-r0/temp/run.deploy_image_license_manifest.3114489', 1, None, None)
> ERROR: Logfile of failure stored in: /home/ubuntu/balena-raspberrypi/build/tmp/work/revpi_connect_s-poky-linux/balena-image/1.0-r0/temp/log.do_image_complete.3114489
> Log data follows:
> | DEBUG: Executing python function sstate_task_prefunc
> | DEBUG: Python function sstate_task_prefunc finished
> | DEBUG: Executing python function do_image_complete
> | NOTE: Executing deploy_image_license_manifest ...
> | DEBUG: Executing shell function deploy_image_license_manifest
> | cp: cannot stat '/home/ubuntu/balena-raspberrypi/build/tmp/deploy/licenses/balena-image-revpi-connect-s-20230120061525/image_license.manifest': No such file or directory
> | WARNING: exit code 1 from a shell command.
> | DEBUG: Python function do_image_complete finished
> ERROR: Task (/home/ubuntu/balena-raspberrypi/build/../layers/meta-balena/meta-balena-common/recipes-core/images/balena-image.bb:do_image_complete) failed with exit code '1'
> NOTE: Tasks Summary: Attempted 4447 tasks of which 4413 didn't need to be rerun and 1 failed.
> ```

When you get this type of error you can just clean the balena-image target and then build the same target again and it should work.

**Comment by floion** [comment]
Hi @wickywaka can you address the comments I left and also rebase the PR please?
Thank you

File: layers/meta-balena-raspberrypi/recipes-kernel/picontrol/picontrol/0001-Use-modules_install-as-wanted-by-yocto.patch (hunk 2 above)
Comment (wickywaka) [reply layers/meta-balena-raspberrypi/recipes-kernel/picontrol/picontrol/0001-Use-modules_install-as-wanted-by-yocto.patch:21]: @floion It looks this is needed. It fails to compile when I disable remove this patch in the `picontrol.bb`. It was added here https://github.com/balena-os/balena-raspberrypi/commit/04d56a3b7ca3716b961e08b9f2b06876315eaf8c

The error it gives when I disable it is follows:
```
ERROR: Task (/home/ubuntu/balena-raspberrypi/build/../layers/meta-balena-raspberrypi/recipes-kernel/picontrol/picontrol.bb:do_install) failed with exit code '1'
```

**Comment by wickywaka** [comment]
> Hi @wickywaka can you address the comments I left and also rebase the PR please? Thank you

Done. I think it should be ready now.

**Comment by obaidch23** [comment]
Any progress with the kernel update?

Thinking about building a new custom image for the kunbus revpi-connect-s, so I wanted to be sure which kernel version should I be using.

PS: Already started working on it with the kernel v5.10, just waiting for review.

here's a link for the kinbus revpi-connect-s board support
https://github.com/balena-os/balena-raspberrypi/pull/913#

**Comment by wickywaka** [comment]
> Any progress with the kernel update?
> 
> Thinking about building a new custom image for the kunbus revpi-connect-s, so I wanted to be sure which kernel version should I be using.
> 
> PS: Already started working on it with the kernel v5.10, just waiting for review.
> 
> here's a link for the kinbus revpi-connect-s board support #913

As far as I can tell this one is ready but we need someone from balena to merge it.

**Comment by floion** [comment]
Closing this PR in favor of https://github.com/balena-os/balena-raspberrypi/pull/953

**Pull request closed without merging**